pkg/watch/watch.go

```diff
@@ -318,27 +324,64 @@ func (o *WatchClient) eventWatcher(ctx context.Context, sourcesWatcher *fsnotify
 	retryTimer := time.NewTimer(time.Millisecond)
 	<-retryTimer.C
 
+	// List of pod phases that will run as background processes
 	podsPhases := NewPodPhases()
 
 	for {
+
 		select {
 		case event := <-sourcesWatcher.Events:
 			events = append(events, event)
 			// We are waiting for more events in this interval
 			sourcesTimer.Reset(100 * time.Millisecond)
+
+			// Trigger manual sync
+		case <-manualSync:
+			log.Section("Manually syncing files")
+			log.Print("Updating component ...")
+
+			// Before we push a sync, we *should* still check to see what files have changed
+			changedFiles, deletedPaths := evaluateChangesHandler(events, parameters.Path, parameters.FileIgnores, sourcesWatcher)
+
+			// Sync files REGARDLESS if changedFiles / deletedPaths == 0
+			// Set the state to sync being outdated
+			componentStatus.State = StateSyncOutdated
+
+			// Push the files
+			log.Print("Pushing files ...")
+			retry, err := processEventsHandler(changedFiles, deletedPaths, parameters, out, &componentStatus, expBackoff)
+			if err != nil {
+				return err
+			}
+
+			// Empty all events
+			if componentStatus.State == StateReady {
+				events = []fsnotify.Event{} // empty the events slice to capture new events
+			}
+
+			// Reset the retry timer
+			if retry != nil {
+				retryTimer.Reset(*retry)
+			} else {
+				retryTimer.Reset(time.Millisecond)
+				<-retryTimer.C
+			}
```

Comment:
Ths code is very similar to the one in `case <-sourcesTimer.C:`

We should try to create a function with this code.